docs/general/build/smart-contracts/contract-security/contract-security.md

```diff
@@ -0,0 +1,35 @@
+---
+title: Smart Contract Security
+displayed_sidebar: generalSidebar
+---
+
+Ensuring the security of smart contracts is crucial as they directly handle and store value, and are difficult to modify once deployed on a blockchain. To secure your smart contracts, follow these key steps and best practices:
+
+### Comprehensive Testing and Auditing
+- **Unit Testing**: Write tests for every function in your contract to ensure it works as expected under various conditions.
+- **Integration Testing**: Test the interactions between contracts.
+- **Fuzz Testing**: Use randomly generated input data to test your contracts, looking for vulnerabilities.
+- **Security Audits**: Conducted by professional third parties, these audits scrutinize your contract's code for security vulnerabilities and poor programming practices.
+
+### Adhere to Known Best Practices
+- **Limit Permissions**: Ensure only authorized users can execute critical functions.
+- **Validate Inputs and Use Assertions**: Verify external inputs and contract states to ensure they meet expectations.
+- **Prevent Reentrancy Attacks**: Use mutexes (e.g., state variables) to prevent multiple calls within the same transaction.
+- **Use Verified Libraries and Contracts**: Reduce errors and vulnerabilities by using open-source libraries and contracts that have been extensively tested and audited.
+
+### Avoid Common Security Pitfalls
+- **Integer Overflow and Underflow**: Use safe math libraries to prevent these issues.
+- **Runtime Checks in Frontend**: Monitoring contract transactions through the frontend application can help detect abnormal behaviors early.
```

Comment:
Indeed, frontend checks are not a panacea, but by monitoring transactions in the frontend, you can quickly identify unusual patterns or suspicious activities. 

Frontend checks serve as an additional layer of security, complementing the internal mechanisms of smart contracts.

Reply:
That makes sense. But I think we should mention it in the article
```suggestion
- **Runtime Checks in Frontend**: Monitoring contract transactions through the frontend application can help detect abnormal behaviors early. This will serve as an additional layer of security, helping to quickly identify unusual patterns or suspicious activities, but it should also be noted that attackers always have the ability to directly call your contract.
```